metatensor-torch/src/atomistic/system.cpp

```diff
@@ -513,6 +546,38 @@ void SystemHolder::set_cell(torch::Tensor cell) {
     this->cell_ = std::move(cell);
 }
 
+void SystemHolder::set_pbc(torch::Tensor pbc) {
+    if (pbc.device() != this->device()) {
```

Comment:
should we move the constructor checks in this function & then call this function from constructor to perform the same checks? Right now this is missing the check for cell vectors vs pbc

Reply:
Moving all the checks into the setter meant dropping a lot of detail in the error messages. I've re-written the check in the setter so the error messages can be better